plugins/filter_aws/aws.c

```diff
@@ -427,8 +429,113 @@ static int get_ec2_tag_values(struct flb_filter_aws *ctx)
     return 0;
 }
 
+static int tag_is_present_in_list(struct flb_filter_aws *ctx, flb_sds_t tag,
+        const char *tags)
+{
+    const char *tag_start;
+    const char *tag_end;
+    const char *tag_sep_before;
+    const char *tags_start;
+    const char *tag_sep_after;
+    const char *tags_end;
+
+    tag_start = strstr(tags, tag);
+    if (!tag_start) {
+        return FLB_FALSE;
+    }
+    tag_end = tag_start + strlen(tag) - 1;
+
+    /* if substring exists it not necessarily implies that tag is included */
+    /* tag name must be exactly equal to the value between separators ',' */
+    /* f.e. without additional conditions, substring would accept Name for a,Namee,b */
+
+    tags_start = tags;
+    tags_end = tags + strlen(tags) - 1;
+
+    /* calculate pointers to characters directly before/after tag */
+    /* if tag is in the middle, then these pointers should be ',' */
+    tag_sep_before = (char*)(tag_start - 1);
+    tag_sep_after = (char*)(tag_end + 1);
+    if (tag_sep_before < tags_start) {
+        tag_sep_before = NULL;
+    }
+    if (tag_sep_after > tags_end) {
+        tag_sep_after = NULL;
+    }
+
+    /* tag is present only if both of the following conditions are true */
+    /* 1. tag is the first one || ',' is directly before the tag */
+    /* 2. tag is the last one  || ',' is directly after the tag */
+    /* code below checks these conditions, but using negative cases */
+    if (tag_start != tags_start && (tag_sep_before && *tag_sep_before != ',')) {
+        return FLB_FALSE;
+    }
+    if (tag_end != tags_end && (tag_sep_after && *tag_sep_after != ',')) {
+        return FLB_FALSE;
+    }
+
+    return FLB_TRUE;
+}
+
+static int get_ec2_tag_enabled(struct flb_filter_aws *ctx)
+{
+    const char *tags_include;
+    const char *tags_exclude;
+    int i;
+    int tag_present;
+
+    /* if there are no tags, there is no need to evaluate which tag is enabled */
+    if (ctx->tags_count == 0) {
+        return 0;
+    }
+
+    /* allocate memory for 'tag_is_enabled' for all tags */
+    ctx->tag_is_enabled = flb_calloc(ctx->tags_count, sizeof(int));
+    if (!ctx->tag_is_enabled) {
+        flb_plg_error(ctx->ins, "Failed to allocate memory for tag_is_enabled");
+        return -1;
+    }
+
+    /* if tags_include and tags_exclude are not defined, set all tags as enabled */
+    for (i = 0; i < ctx->tags_count; i++) {
+        ctx->tag_is_enabled[i] = FLB_TRUE;
+    }
+
+    /* apply tags_included configuration */
+    tags_include = flb_filter_get_property("tags_include", ctx->ins);
+    if (tags_include) {
+        for (i = 0; i < ctx->tags_count; i++) {
+            tag_present = tag_is_present_in_list(ctx, ctx->tag_keys[i], tags_include);
```

Comment:
@mwarzynski this code does not seem ideal to me. I think it'd be simpler and more readable and maintainable to first, iterate over the `tags_include` and split on separate and create a tags include list. And then you have two lists, one of the actual tag keys, one of hte tags you want to include. And then you can do simple logic iterating over the list.

Reply:
I've changed the implementation. Would you please take a look?

Although it's a better solution, I am not convinced my current implementation is more readable and maintainable. That's on me. I think there should be a more readable implementation though. Feel free to give more precise suggestions how to proceed with improvements.

Reply:
I like the new approach